public/app/features/alerting/unified/state/actions.ts

```diff
@@ -292,8 +327,11 @@ export const fetchRulesSourceBuildInfoAction = createAsyncThunk(
 
 export function fetchAllPromAndRulerRulesAction(force = false): ThunkResult<Promise<void>> {
   return async (dispatch, getStore) => {
+    const allStartLoadingTs = performance.now();
```

Comment:
In other cases you abstracted the performance code within the `logInfo` function, was it not possible to do something similar in this case?

Reply:
It's technically possible but I think readability would suffer significantly. I'd have to wrap `Promise.allSettled` into a function wrapped into `withPerfLoggin`. I think it's not worth it